lib-rpc-server/ocamlformat_rpc.ml

```diff
@@ -65,7 +65,7 @@ let run_config conf c =
 let run_path path =
   match
     Bin_conf.build_config ~enable_outside_detected_project:false ~root:None
-      ~file:path ~is_stdin:false
+      ~file:path ~is_stdin:false ~on_save:true
```

Comment:
We might want an option in the protocol to change this (maybe `true` by default though).

Reply:
I'm trying to think of a case where we don't want the RPC client (which are editors, dune doesn't use the rpc but directly calls ocamlformat) to use `on_save = true`, it sounds like a bad idea?

Reply:
I think the RPC client definitely wants `on_save = false`. It might be simpler if the RPC options has the same defaults as the main tool indeed.

Reply:
@rgrinberg suggested that the RPC could have a new error case for when the file hasn't been formatted due to no detected project.

Seems more interesting than returning the input unchanged to me.